pkg/webhook/validating/webhook.go

```diff
@@ -198,6 +202,29 @@ func (a *AdmissionHandler) validateManagedCluster(request *v1.AdmissionRequest)
 		}
 		// the clusterdeployment is not claimed from clusterpool, allow to update clustersetlabel.
 		return a.validateAllowUpdateClusterSet(request.UserInfo, oldClusterSet, newClusterSet)
+	case v1.Delete:
+		clusterName := request.Name
+		clusters, err := a.ClusterLister.List(labels.Everything())
+		if err != nil {
+			return a.responseNotAllowed(err.Error())
+		}
+
+		var hostedClusters = make([]string, 0)
+		for _, cluster := range clusters {
+			if cluster.GetAnnotations()[constants.AnnotationKlusterletDeployMode] != "Hosted" {
+				continue
+			}
+			if cluster.GetAnnotations()[constants.AnnotationKlusterletHostingClusterName] != clusterName {
+				continue
+			}
+			hostedClusters = append(hostedClusters, cluster.Name)
+		}
+
+		if len(hostedClusters) > 0 {
```

Comment:
@zhujian7 I don't understand how this works. If there are multiple hosting clusters, how do you get the list of hosted clusters for a specific hosting cluster that the user chooses to delete?

Reply:
there is a condition check `cluster.GetAnnotations()[constants.AnnotationKlusterletHostingClusterName] != clusterName` on line 217.

Reply:
Ah OK Thanks @zhujian7